x/projects/keeper/migrations.go

```diff
@@ -25,3 +27,71 @@ func (m Migrator) Migrate2to3(ctx sdk.Context) error {
 	}
 	return nil
 }
+
+// Migrate3to4 implements protobuf migration from v3 to v4:
+// Now the project protobuf has admin and subscription policies
+func (m Migrator) Migrate3to4(ctx sdk.Context) error {
+	projectIndices := m.keeper.projectsFS.GetAllEntryIndices(ctx)
+	for _, projectIndex := range projectIndices {
+		blocks := m.keeper.projectsFS.GetAllEntryVersions(ctx, projectIndex, true)
+		for _, block := range blocks {
+			var oldProjectStruct v3.Project
+			if found := m.keeper.projectsFS.FindEntry(ctx, projectIndex, block, &oldProjectStruct); !found {
+				return fmt.Errorf("could not find project with index %s", projectIndex)
+			}
+
+			// convert project keys from type v3.ProjectKey to types.ProjectKey
+			newProjectKeys := []types.ProjectKey{}
+			for _, oldProjectKey := range oldProjectStruct.ProjectKeys {
+				newProjectKey := types.ProjectKey{
+					Key:   oldProjectKey.Key,
+					Vrfpk: oldProjectKey.Vrfpk,
+				}
+
+				for _, oldprojectKeyType := range oldProjectKey.Types {
+					if oldprojectKeyType == v3.ProjectKey_ADMIN {
+						newProjectKey.Types = append(newProjectKey.Types, types.ProjectKey_ADMIN)
```

Comment:
Why the switch back from bitmap (of key types) to inefficient list?  (@Yaroms)

Reply:
bitmap was never implemented